src/compiler/program.ts

```diff
@@ -1430,12 +1430,10 @@ namespace ts {
             return a.fileName === b.fileName;
         }
 
-        function moduleNameIsEqualTo(a: LiteralExpression, b: LiteralExpression): boolean {
-            return a.text === b.text;
-        }
-
-        function getTextOfLiteral(literal: LiteralExpression): string {
-            return literal.text;
+        function moduleNameIsEqualTo(a: StringLiteral | Identifier, b: StringLiteral | Identifier): boolean {
+            return a.kind === SyntaxKind.StringLiteral

```

Comment:
Just to confirm, the string literal `"global"` is not equal to the identifier `global`?

Reply:
```ts
export {}; // this is a module

// Augments a module named "global"
declare module "global" {
    export const x: number;
}

// Augments the global scope
declare global {
    const y: number;
}

// Module access: "y" fails
import { x as _x, y as _y } from "global";

// Global access: "x" fails
x; y;
```